test/jdk/sun/security/pkcs11/Cipher/TestChaChaPoly.java

```diff
@@ -0,0 +1,216 @@
+/*
```

Comment:
There is no compatibility Test exist between SunJCE and SunPKCS11 providers. Do we need one here.

Reply:
Theoretically, these new regression tests are mostly adapted from the ChaCha20-Poly1305 cipher impl of existing SunJCE provider. The test vectors are also the same, so the compatibility is tested indirectly. I also added a testInterop() method which uses one for encryption and the other for decryption and vice versa, just to be safe. If there are specific scenarios which I missed, please let me know and I will add it.